File: archive/defunct_20250926_090506/mlb_backend/scripts/shared/timeUtilsBackend.js
// backend/scripts/shared/timeUtilsBackend.js

import { DateTime } from "luxon";

// ✅ Add previously ported functions here...
export function nowET() {
  return DateTime.now().setZone("America/New_York");
}

export function todayET() {
  return nowET().toISODate();
}

export function yesterdayET() {
  return nowET().minus({ days: 1 }).toISODate();
}

export function currentTimeET() {
  return nowET().toFormat("HH:mm");
}

export function toISODate(dateInput) {
  if (!dateInput) return null;

  if (typeof dateInput === "string") {
    return DateTime.fromISO(dateInput).toISODate();
  } else if (dateInput instanceof Date) {
    return DateTime.fromJSDate(dateInput).toISODate();
  } else if (DateTime.isDateTime(dateInput)) {
    return dateInput.toISODate();
  }

  console.warn("⚠️ Invalid date input provided to toISODate:", dateInput);
  return null;
}

// ✅ Add these two exports for MLB API support
export async function getGameStartTimeET(gameId) {
  try {
    let res = await fetch(
      `https://statsapi.mlb.com/api/v1/schedule?sportId=1&gamePk=${gameId}`
    );
    if (res.ok) {
      const schedJson = await res.json();
      const schedISO = schedJson?.dates?.[0]?.games?.[0]?.gameDate;
      if (schedISO) {
        return DateTime.fromISO(schedISO).setZone("America/New_York").toISO();
      }
    }

    res = await fetch(
      `https://statsapi.mlb.com/api/v1/game/${gameId}/boxscore`
    );
    if (res.ok) {
      const boxJson = await res.json();
      const boxISO = boxJson?.gameData?.datetime?.dateTime;
      if (boxISO) {
        return DateTime.fromISO(boxISO).setZone("America/New_York").toISO();
      }
    }

    return null;
  } catch (err) {
    console.warn(`⚠️ Could not get game time for ${gameId}:`, err.message);
    return null;
  }
}

export function getTimeOfDayBucketET(isoDateTime) {
  const hour = DateTime.fromISO(isoDateTime, { zone: "America/New_York" }).hour;
  if (hour < 12) return "morning";
  if (hour < 17) return "afternoon";
  if (hour < 21) return "evening";
  return "night";
}

export function getDayOfWeekET(isoDate) {
  return DateTime.fromISO(isoDate, { zone: "America/New_York" }).toFormat(
    "cccc"
  );
}

// ✅ Combine gameDate and gameTime into an Eastern-zone DateTime object
export function toEasternDateTime(gameDate, gameTime) {
  if (!gameDate || !gameTime) return null; // guard clause
  const iso = `${gameDate}T${gameTime}`; // e.g. 2023-04-30T19:05:00
  return DateTime.fromISO(iso, { zone: "America/New_York" });
}

File: backend/mlb/mlb/shared/time_utils_backend.py
# File: backend/scripts/shared/time_utils_backend.py

from datetime import datetime
from typing import Optional
from pytz import timezone
import requests
from dateutil import parser

ET = timezone("US/Eastern")

def now_et() -> datetime:
    return datetime.now(ET)

def today_et() -> str:
    return now_et().strftime("%Y-%m-%d")

def yesterday_et() -> str:
    return (now_et() - timedelta(days=1)).strftime("%Y-%m-%d")

def format_game_time(iso_datetime: str) -> dict:
    if not iso_datetime:
        return {"etTime": "", "localTime": ""}
    dt = parser.isoparse(iso_datetime)
    et_time = dt.astimezone(ET).strftime("%H:%M")
    local_time = dt.strftime("%H:%M")
    return {"etTime": et_time, "localTime": local_time}

def format_date_et(date_string: str) -> str:
    dt = parser.isoparse(date_string).astimezone(ET)
    return dt.strftime("%b %d, %Y")

def get_day_of_week_et(iso_date: str) -> str:
    dt = parser.isoparse(iso_date).astimezone(ET)
    return dt.strftime("%A")

def get_time_of_day_bucket_et(iso_datetime):
    if isinstance(iso_datetime, str):
        dt = parser.isoparse(iso_datetime)
    elif isinstance(iso_datetime, datetime):
        dt = iso_datetime
    else:
        raise ValueError(f"Unsupported datetime input: {iso_datetime}")

    hour = dt.astimezone(ET).hour

    if hour < 12:
        return "morning"
    elif hour < 17:
        return "afternoon"
    elif hour < 20:
        return "evening"
    else:
        return "night"
def get_game_start_time_et(game_id: int) -> Optional[str]:
    try:
        sched_url = f"https://statsapi.mlb.com/api/v1/schedule?sportId=1&gamePk={game_id}"
        sched_res = requests.get(sched_url).json()
        sched_iso = sched_res.get("dates", [{}])[0].get("games", [{}])[0].get("gameDate")
        if sched_iso:
            return parser.isoparse(sched_iso).astimezone(ET).isoformat()
        box_url = f"https://statsapi.mlb.com/api/v1/game/{game_id}/boxscore"
        box_res = requests.get(box_url).json()
        box_iso = box_res.get("gameData", {}).get("datetime", {}).get("dateTime")
        if box_iso:
            return parser.isoparse(box_iso).astimezone(ET).isoformat()
    except Exception as e:
        print(f"⚠️ Could not get game time for {game_id}: {e}")
    return None
